data_steward/analytics/cdr_ops/cross_ct_rt/dst_251_addtional_QCs_for_the_Controlled_Tier_part1.py

```diff
@@ -0,0 +1,970 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.7.1
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# # DST-251 Add more QA/QCs for the Controlled Tier data
+#
+# We want to mirror the QCs in here   to add more checks to the Controlled Tier (CT).
+#
+# The privacy checks against the CT are done. But we need more QCs that check against the OMOP rules AND against the Program constraints. 
+#
+# Create a notebook for the following checks AND commit them here. If you are just starting working WITH curation, follow the instructions here  .
+#
+# The checks are:
+#
+# 1 No dates earlier than birth dates
+#
+# 2 No dates earlier than 1980 in any table, except for the observation
+#
+# 3 No dates after death:done
+#
+# 4 No WITHdrawn participants: need pdr access
+#
+# 5 No data after participant's suspension: need pdr access
+#
+# 6 All participants have basics
+#
+# 7 All participants WITH EHR data have said yes to EHR consents
+#
+# 8 All participants WITH Fitbit have said yes to primary consents
+#
+# 9 All primary keys are in _ext
+#
+# 10 No duplicated primary keys
+#
+# 11 OMOP tables should have standard concept ids
+#
+# 12 observation concept ids (4013886, 4135376, 4271761) that have dates equal to birth dates should be set to CDR cutoff date
+#
+# 13 all other observation concept ids WITH dates similar to birth dates other than the 3 above should be removed
+#
+# 14 All the descendants of ancestor_concept_id IN (4054924, 141771) -- motor vehicle accidents should be dropped in condition_occurrence table
+
+# + tags=["parameters"]
+# Parameters
+project_id = ""
+rt_dataset = ""
+ct_dataset = ""
+earliest_ehr_date=""
+cut_off_date=""
+
+# +
+import pandas as pd
+from analytics.cdr_ops.notebook_utils import execute
+from common import JINJA_ENV
+from utils import auth
+from utils.bq import get_client
+
+client = get_client(project_id)
+
+pd.options.display.max_rows = 120
+# -
+
+# summary will have a summary in the end
+df = pd.DataFrame(columns = ['query', 'result']) 
+
+# # Query1: No dates before birth_date
+#
+
+# +
+# step1 , to get the tables AND columns that have person_id, size >1 AND DATE columns AND save to a data frame
+query = JINJA_ENV.from_string("""
+
+WITH
```

Comment:
@jqian2015 is there a reason why TIMESTAMP columns are not included in this list of collumns?

Reply:
i thought TMESTAMP is same as date column but with extra 'timestamp' and since we only use 'date' , so I did not use it. if this assumption is wrong, we can sure add TIMESTAMP.